### PR TITLE
Single selection of jurisdictions for FI and Dynamic FI

### LINKED ISSUE #988: Add support for single jurisdiction selection on Jurisdiction Assignment Page
For certain plan intervention types e.g. FI and Dynamic-FI, you are only allowed to have one jurisdiction selected when creating the plan.

This jurisdiction must be a leaf-node in the jurisdiction tree.

Part of: https://github.com/onaio/reveal-frontend/issues/986

### PR DESCRIPTION
**Pull request by gmwasi**
Add support for single jurisdiction selection on Jurisdiction Assignment Page
resolves #988

### CODE REVIEW

File: src/constants.tsx [hunk 1]
```diff
@@ -232,3 +232,7 @@ export const APPLICATION_X_CSV = 'application/x-csv';
 export const TEXT_COMMA_SEPARATED_VALUES = 'text/comma-separated-values';
 export const TEXT_X_COMMA_SEPARATED_VALUES = 'text/x-comma-separated-values';
 export const TEXT_TAB_SEPARATED_VALUES = 'text/tab-separated-values';
+
+/** Intervention Types */
+export const INTERVENTION_TYPE_FI = 'FI';
+export const INTERVENTION_TYPE_DYNAMIC_FI = 'Dynamic-FI';
```
Comment (moshthepitt) [inline src/constants.tsx:238]: Please get these from here https://github.com/onaio/reveal-frontend/blob/b35bb12cb247bb916d1a1b0f29204a38cea56ec7/src/store/ducks/plans.ts#L15 instead

File: src/containers/pages/JurisdictionAssignment/JurisdictionTable/tests/__snapshots__/index.test.tsx.snap [hunk 2]
```diff
@@ -3,6 +3,7 @@
 exports[`src/containers/pages/jurisdictionView/jurisdictionTable selects and deselect nodes: should be now checked 1`] = `
 <input
   checked={true}
+  disabled={true}
```
Comment (moshthepitt) [inline src/containers/pages/JurisdictionAssignment/JurisdictionTable/tests/__snapshots__/index.test.tsx.snap:6]: Could we add a test that shows this as true:

- For FI/Dynamic-FI
- For other plan types

File: src/containers/pages/JurisdictionAssignment/JurisdictionTable/index.tsx [hunk 3]
```diff
@@ -112,7 +112,23 @@ const JurisdictionTable = (props: JurisdictionSelectorTableProps) => {
     }
   }
 
+  const isSingleSelect = () => {
```
Comment (moshthepitt) [inline src/containers/pages/JurisdictionAssignment/JurisdictionTable/index.tsx:115]: Lets add a docstring for this one

**Comment by Naima-Bashir** [comment]
How to test: create a dynamic plan, test the checkbox selection
: test if IRS checkbox works
For certain plan intervention types e.g. FI and Dynamic-FI, you are only allowed to have one jurisdiction selected when creating the plan.
This jurisdiction must be a leaf-node in the jurisdiction tree.
disable checkbox for intervention types other than IRS.

**Comment by Naima-Bashir** [comment]
FI: Checkbox cant be clicked: image
IRS: One can select the checkbox 
image

**Pull request merged**